map-low.go

```diff
@@ -416,22 +423,23 @@ func (m *BPFMapLow) UpdateBatch(keys, values unsafe.Pointer, count uint32) error
 		optsC,
 	)
 	errno = syscall.Errno(-retC)
-	if retC < 0 {
-		if errno != syscall.EFAULT && uint32(countC) != count {
-			return fmt.Errorf("failed to update ALL elements in map %s, updated (%d/%d): %w", m.Name(), uint32(countC), count, errno)
-		}
-		return fmt.Errorf("failed to batch update elements in map %s: %w", m.Name(), errno)
+	// retC < 0 && errno == syscall.E2BIG indicates a partial update.
```

Comment:
Is the comment out of date? (`!=` rather than `==`)?

Reply:
I don't think so, it states that it can be updated partially until getting E2BIG. Perhaps it can be better explained. WDYT?

Reply:
It's mostly a nit, but I find it confusing that this comment is here when it's dealing with another case. But again, it's a personal style preference, I am not super strong about this one!

Reply:
Possibly it would be better as:

```c
// errno == syscall.E2BIG may indicate a partial update, so only consider it as an error when (errno != syscall.E2BIG || countC == 0).
```

The same pattern is valid to the other function comments. WDYT?